src/ToolingSharedLibrary/Includes/VbsEnclaveABI/Shared/ConversionHelpers.h

```diff
@@ -0,0 +1,446 @@
+        {
+            return ConvertStruct<DecayedTarget>(src);
+        }
+        else if constexpr (AreBothEnums<DecayedSrc, DecayedTarget>)
+        {
+            return static_cast<DecayedTarget>(src);
+        }
+        // Convert WStringT flatbuffer src to std::wstring target. 
+        else if constexpr (AreBothTheSame<DecayedTarget, std::wstring>)
+        {
+            // The Codegen will only pass std::wstring as a target when the source is related to WStringT.
+            return ConvertToStdWString(src);
+        }
+        // Convert std::wstring source to WStringT flatbuffer target.
+        else if constexpr (AreBothTheSame<DecayedSrc, std::wstring>) 
+        {
+            // The Codegen will only pass std::wstring as a source when the target is related to WStringT.
+            return CreateWStringT<DecayedTarget>(src);
+        }
+        else if constexpr (AreBothUniquePtrs<DecayedSrc, DecayedTarget>)
+        {
+            return ConvertToUniquePtr<DecayedTarget>(src);
+        }
+        else if constexpr (UniquePtr<DecayedTarget> && Optional<DecayedSrc>)
+        {
+            return ConvertToUniquePtr<DecayedTarget>(src);
+        }
+        else if constexpr (UniquePtr<DecayedSrc> && NonWrappedStruct<DecayedTarget>)
+        {
+            if (!src)
+            {
+                return {};
+            }
+
+            using InnerSrcType = unique_ptr_inner_type_t<DecayedSrc>;
+            return ConvertType<DecayedTarget>(*src);
+        }
+        else if constexpr (NonWrappedStruct<DecayedSrc> && UniquePtr<DecayedTarget>)
+        {
+            return ConvertToUniquePtr<DecayedTarget>(src);
+        }
+        else if constexpr (RawPtr<DecayedSrc> && UniquePtr<DecayedTarget>)
+        {
+            return ConvertToUniquePtr<DecayedTarget>(src);
+        }
+        else if constexpr (IsRawPtrAndNonWrappedStruct<DecayedSrc>)
+        {
+            return ConvertToUniquePtr<DecayedTarget>(src);
+        }
+        // pointers to primitive/enum dev types get converted to std::optional primitive/enum flatbuffers.
+        else if constexpr (IsRawPtrAndNotNonWrappedStruct<DecayedSrc>)
+        {
+            return ConvertToOptional<DecayedTarget>(src);
+        }
+        else if constexpr (UniquePtr<DecayedSrc> && Optional<DecayedTarget>)
+        {
+            return ConvertToOptional<DecayedTarget>(src);
+        }
+        else if constexpr (AreBothVectors<DecayedSrc, DecayedTarget> || (StdArray<DecayedSrc> && Vector<DecayedTarget>))
+        {
+            using InnerSrcType = vector_or_array_inner_type_t<DecayedSrc>;
+            using InnerTargetType = vector_or_array_inner_type_t<DecayedTarget>;
+            return TransformVector<DecayedTarget>(src, [] (const InnerSrcType& value)
+            {
+                return ConvertType<InnerTargetType>(value);
+            });
+        }
+        else if constexpr (Vector<DecayedSrc> && StdArray<DecayedTarget>)
```

Comment:
same here, I don't think this the src is interesting.

        else if constexpr (StdArray<DecayedTarget>)

^ this is enough.